pkg/dockerregistry/server/manifesthandler.go

```diff
@@ -36,36 +36,25 @@ type ManifestHandler interface {
 }
 
 // NewManifestHandler creates a manifest handler for the given manifest.
-func NewManifestHandler(repo *repository, manifest distribution.Manifest) (ManifestHandler, error) {
+func NewManifestHandler(serverAddr string, blobStore distribution.BlobStore, manifest distribution.Manifest) (ManifestHandler, error) {
 	switch t := manifest.(type) {
 	case *schema1.SignedManifest:
-		return &manifestSchema1Handler{repo: repo, manifest: t}, nil
+		return &manifestSchema1Handler{serverAddr: serverAddr, blobStore: blobStore, manifest: t}, nil
 	case *schema2.DeserializedManifest:
-		return &manifestSchema2Handler{repo: repo, manifest: t}, nil
+		return &manifestSchema2Handler{blobStore: blobStore, manifest: t}, nil
 	default:
 		return nil, fmt.Errorf("unsupported manifest type %T", manifest)
 	}
 }
 
-// NewManifestHandlerFromImage creates a new manifest handler for a manifest stored in the given image.
-func NewManifestHandlerFromImage(repo *repository, image *imageapiv1.Image) (ManifestHandler, error) {
-	var (
-		manifest distribution.Manifest
-		err      error
-	)
-
+// NewManifestFromImage creates a manifest for a manifest stored in the given image.
+func NewManifestFromImage(image *imageapiv1.Image) (distribution.Manifest, error) {
```

Comment:
```go
if len(image.DockerImageManifest) == 0 {
	return nil, fmt.Errorf("unable to create manifest for %s image: empty field", image.Name)
}
```
maybe better to add check to avoid unmarshal errors ?

Reply:
It's never called with a such precondition.

Reply:
Though, fixed.